lib/modules/manager/rpm/extract.ts

```diff
@@ -1,5 +1,35 @@
 import { logger } from '../../../logger';
 import type { PackageFileContent } from '../types';
+import { parseSingleYaml } from '../../../util/yaml';
+import { readLocalFile, localPathExists } from '../../../util/fs';
+import { RedHatRPMLockfile } from './schema';
+import type { RedHatRPMLockfileDefinition } from './schema';
+import type { PackageDependency } from '../types';
+import { exec } from '../../../util/exec';
+import type { ExecOptions } from '../../../util/exec/types';
+
+async function getUpdatedLockfile() {
+  const cmd: string[] = [];
+  const packageFileName = "rpms.in.yaml";
+  const outputName = "rpms.lock.tmp.yaml";
+
+  if (await localPathExists(outputName)) {
+    // Only generate the temporary lockfile once
+    return;
+  }
+
+  cmd.push(`rpm-lockfile-prototype ${packageFileName} --outfile ${outputName}`);
```

Comment:
We're running `rpm-lockfile-prototype` twice (another one is in `artifacts.ts`. For some repositories, running the script can be resource and time consuming, do you think we can run it once in each task? After extracting the rpms from the original lockfile, we don't need the original lockfile anymore, how about just overriding the original lockfile?